src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -1158,6 +1175,17 @@ private static TimeSpan ParseTimeSpan(string name, string value)
             }
         }
 
+        private bool EnsureTlsDisableCertificateRevocationCheckIsValid(bool value)
+        {
+            if (_tlsDisableCertificateRevocationCheck.HasValue && _tlsDisableCertificateRevocationCheck.Value != value)
+            {
+                var name = nameof(_tlsDisableCertificateRevocationCheck).Substring(1);
```

Comment:
It looks good, should we do it everywhere?

Reply:
Thanks Dima! I think this pattern may be helpful in the future as well. I'd prefer not to make the changes to other places as it feels out of scope for this PR